### DISCUSSION TITLE
Difference between webcontainer and browser vs code extensions?

### DISCUSSION THREAD

**Discussion by Readf0x** [stackblitz/core#2842] [Q&A] [answered]
I've noticed that the extensions panel has two installed extensions panels, `Webcontainer - Installed` and `Browser - Installed`. What are the difference between the two? Obviously it specifies where the extension is running, which dictates what it can access, like monitoring the vite server and such. The real question is how do I install extensions in the browser? Or has that not been implemented yet?
![image](https://github.com/stackblitz/core/assets/57838054/f50ada9f-decb-4027-8c3a-9a5c0e23cd82)

**Comment by samdenty** [comment, marked as answer]
This split comes from the move we did a while back to https://code.visualstudio.com/docs/remote/vscode-server#_architecture. We run a webcontainer process with vscode server, and so all extensions in webcontainer are ran inside node.js processes.

The browser extensions i haven't personally been able to install yet, but they are [web worker extensions](https://code.visualstudio.com/api/extension-guides/web-extensions) and run in a more restricted sandbox. They're more lightweight than webcontainer ones and often are made to be specifically browser-compatible - whereas webcontainer ones sometimes can be hit and miss depending on whether they use native linux commands etc. which we don't support

But yeah the split is confusing because everything is running in your browser tab, we might get rid of the browser panel in the UI if it's not required (will have to investigate). Thanks for reminding us
